### PR TITLE
[INT-14388] Add Status component

### PR DESCRIPTION
**Pull request by wyvl**
INT-14388

**The following MUST be checked prior to approval. If not relevant to your PR, remove the line from your description.**
- [x]  The `styleguide.config.js` has been updated to show examples/new functionality for the component
- [x]  Test Coverage for any new or updated functionality
- [ ]  New and updated components have been tested locally in lapis and SPA and work as expected
- [x]  This PR has been tagged with PATCH, MINOR, or MAJOR.
- [x]  The component has data-component-type and data-component-context attributes following the standard
- [ ]  You have requested a cross-team review on this component so everyone knows it exists

### CODE REVIEW

**Comment by wyvl** [comment]
Screen Shot 2020-10-08 at 5 30 22 pm
Screen Shot 2020-10-08 at 5 30 28 pm

File: src/domain/Statuses/Status/Status.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+enum Size {
+  Small = 'small',
+  Large = 'large'
+}
+
+interface IStatusProps {
+  /** Component context */
+  componentContext?: string
+  /** What style variant of status dot to display */
+  variant?: StatusDotVariants
```
Comment (andrewhillcode) [inline src/domain/Statuses/Status/Status.tsx:17]: I think this should be required, there isnt really a situation where you don't want to pass in a variant anyway

**Comment by wyvl** [comment]
Screen Shot 2020-10-08 at 5 51 32 pm

File: src/domain/Statuses/Status/Status.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+
+import { Props, Variables } from '../../../common'
+import { Text } from '../../Typographies/Text'
+import { StatusDot, StatusDotVariants } from '../StatusDot'
+import { StyledStatus } from './style'
+
+enum Size {
+  Small = 'small',
+  Large = 'large'
```
Comment (andrewhillcode) [inline src/domain/Statuses/Status/Status.tsx:10]: should this be normal/regular/default/medium? whatever the word we use for that size of typography

File: src/domain/Statuses/StatusDot/style.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  [StatusDotVariants.Warning]: Variables.Color.y400,
+  [StatusDotVariants.Success]: Variables.Color.g400,
+  [StatusDotVariants.Default]: Variables.Color.n400,
+  [StatusDotVariants.Info]: Variables.Color.i400
+}
+
+const StyledStatusDot = styled.div`
+  width: ${Variables.Spacing.sMedium}px;
+  height: ${Variables.Spacing.sMedium}px;
+  border-radius: ${Variables.Spacing.sMedium}px;
```
Comment (andrewhillcode) [inline src/domain/Statuses/StatusDot/style.ts:31]: The size of the dot shouldn't rely on Spacing variables?

File: src/domain/Statuses/Status/Status.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+interface IStatusProps {
+  /** Component context */
+  componentContext?: string
+  /** What style variant of status dot to display */
+  variant?: StatusDotVariants
+  /** The margins around the component */
+  margins?: Props.Margin
+  /** label of the status */
+  children: string
```
Comment (andrewhillcode) [inline src/domain/Statuses/Status/Status.tsx:21]: wonder if we'll encounter any weird issues with this. do you think children is better to use than a `label` prop?

**Comment by wyvl** [comment]
will need to remove isInline on status props

**Review by andrewhillcode — APPROVED** [review]

**Pull request merged**